doc: doxygen: fix incorrect @retval -errno example in guidelines (#107458)
zephyrproject-rtos/zephyr: doc/contribute/style/doxygen.rst

    * @param buf Data to transmit
    * @param size Number of bytes to transmit
    *

>     * @return Number of bytes placed in @p pipe, negative error code on failure.

Reviewer: ```suggestion
    * @return Number of bytes placed in @p pipe, negative errno value on failure.
```

I personally think that omitting `errno` removes some valuable context, and we should refer to errno

Author: @kartben , @carlescufi Could you weigh in on this please?

Reviewer 2: ` * @return -errno code if failure` and similar variations are used across zephyr, errno refers to the `errno.h` header the errors are from, keeping this context is relevant IMO.

Author: @bjarki-andreasen  Which wording exactly would you suggest? The one suggested by @Thalley ?

Reviewer 2: > @bjarki-andreasen Which wording exactly would you suggest? The one suggested by @Thalley ?

Yeah

>     * @return Number of bytes placed in @p pipe, negative errno value on failure.

Reviewer: ```suggestion
    * @return Number of bytes placed in pipe (0 if pipe closed), negative errno value on failure.
```

or change `modem_pipe_transmit` to not have the `(0 if pipe closed)` - Since we are using that function as an example, wouldn't it make sense that the text here and there are identical?

>     * @return Number of bytes placed in @p pipe, or a negative error code on failure.

Reviewer 3: ```suggestion
    * @return Number of bytes placed in @p pipe, or a negative errno error code on failure.
```
We want a specific type of error codes.
lib/libc/minimal/include/errno.h

Author: Makes sense. Should this wording be applied to the function documentation as well, and used across PR #107276?

Reviewer: I think that would make for functions that return an errno value on error

Author: Got it, will update this PR and #107276.

>     * @return Number of bytes placed in @p pipe, or a negative errno error code on failure.

Reviewer: I wonder if `errno error` is unnecessary, and should just be, as `errno` kind of implies error code :p 

```suggestion
    * @return Number of bytes placed in @p pipe, or a negative errno code on failure.
```

Author: I mean, `errno error` reads as `error number error code` which sounds weird to me as well, what do you think @JarmouniA?

Author: @kartben in #65974 suggested: `.., or a negative error code on error.`. So maybe that's the way to go forward with this.

Reviewer: I like the idea of specifying `errno` as that is a specific subset of error codes that you can easily look up, whereas `negative error code` could be anything

Author: Okay, so:
```suggestion
    * @return Number of bytes placed in @p pipe, or a negative errno code on failure.
```

@JarmouniA, @kartben Do you agree?

Reviewer 4: @qbx11 If you search the Zephyr code base, following sentence is used hudreds of times:
```
@return 0 on success, negative error code on failure
```

So use that.

If the API does not specify what error codes are, then those are the ones defined in `errno.h`. Even in POSIX, those are referred as error codes. The `errno` is just a global or thread local variable used for fetching the error code and `errno.h` is a header that list those. So there is no need to use word "errno" when referring to error codes.

Author: Thanks for clarification!

Author: I aligned the changes with your suggestion @SeppoTakalo, could you take a look at this PR and #107276?
Thanks!
